crates/telio-firewall/src/firewall.rs

```diff
@@ -425,6 +446,12 @@ impl StatefullFirewall {
                 "Inbound IP packet is for whitelisted peer, forwarding: {:?}",
                 ip
             );
+
+            // We still need to track the state of TCP connections
+            if matches!(proto, IpNextHeaderProtocols::Tcp) {
```

Comment:
This handle_inbound_tcp function should only be called for whitelisted peers if connection killing is enabled in feature configs, as this would allow as to not pay its performance penalty for platforms where connection killing is not required.

Reply:
Done